src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -510,8 +511,9 @@ private void ScanFormatSpecifier()
             /// <summary>
             /// Scan past the hole inside an interpolated string literal, leaving the current character on the '}' (if any)
             /// </summary>
-            private void ScanInterpolatedStringLiteralHoleBalancedText(char endingChar, bool isHole, ref int colonPosition)
+            private void ScanInterpolatedStringLiteralHoleBalancedText(char endingChar, bool isHole, out Range colonRange)
             {
+                colonRange = default;
```

Comment:
I get the reasoning for the open/close quote range, but why does the colon need a range? Will that be arbitrarily changable?

Reply:
it just kept the logic pretty simple in that:
1. everything is a range.
2. it's easy to represent "not there" just with an empty range (as opposed to needing to use `int` or `range` and then have negative/0 mean 'not there').
3. ranges make slicing super easy.